spec/controllers/accounts/products_controller_spec.rb

```diff
@@ -0,0 +1,180 @@
+require 'rails_helper'
+
+RSpec.describe Accounts::UsersController, type: :request do
+  let!(:account) { create(:account) }
+  let!(:user) { create(:user, account: account) }
+  let(:product) { create(:product, account: account) }
+  let(:product_first) { Product.first }
+
+  describe 'POST /accounts/{account.id}/products' do
+    let(:valid_params) do
+      { product: { name: 'Product name', identifier: 'id123', amount_in_cents: '1500,99', quantity_available: '10',
+                   description: 'Product description', account_id: account.id } }
+    end
+
+    context 'when it is an unauthenticated user' do
+      it 'returns unauthorized' do
+        post "/accounts/#{account.id}/products"
+        expect(response).to redirect_to(new_user_session_path)
+      end
+    end
+
+    context 'when it is an authenticated user' do
+      before do
+        sign_in(user)
+      end
+
+      context 'create product' do
+        it do
+          expect do
+            post "/accounts/#{account.id}/products",
+                 params: valid_params
+          end.to change(Product, :count).by(1)
+          expect(response).to redirect_to(account_products_path(account))
+          expect(product_first.name).to eq('Product name')
+          expect(product_first.identifier).to eq('id123')
+          expect(product_first.amount_in_cents).to eq(150_099)
+          expect(product_first.quantity_available).to eq(10)
+          expect(product_first.description).to eq('Product description')
+          expect(product_first.account_id).to eq(account.id)
+        end
```

Comment:
Para facilitar a leitura =)

```suggestion
        end

```

Reply:
Boa!